sdks/go/pkg/beam/core/runtime/harness/session.go

```diff
@@ -35,18 +33,18 @@ const (
 	dataSend
 )
 
+// This variable is set in the init process when it parses
+// runner options.
+var capture SessionCaptureHook
```

Comment:
(In connection with the high-level comment above) 

Instead of being a global variable, the session writer would be a FooHook implementation and the state below would be local to that instance. Multiple writers thus become possible.